### DISCUSSION TITLE
RFE: Allow Admin to block specific Buildah commands

### DISCUSSION THREAD

**Discussion by TomSweeneyRedHat** [containers/buildah#3276] [Ideas]
In #3270, @sachinkaushik asked if it would be possible to for the administrator to say which Buildah commands could not be run from the CLI.  Per the issue description:

"Hi Team,

This is not issue/bug. I just need your support, I have a requirement where User can only build container image and also run others buildah command on terminal but _User can not run/create container using _ buildah from command. I have to kind of block or disable this command. Exception this command User can run all buildah commands.

As per our project requirement, we are proving a user interface to user he can only run that container image from there only. We have written all backend logic, image will be pulled from OpenShift's internal registry, that container image User has build from terminal and will be pushed to Openshift internal registry.

Any help/guidance will be appreciated.

Thank you in advance..!!!"

We've closed the issue, but I wanted to open up this topic to discuss it further.  This might be a good project for someone looking to discover more about the inner workings of the Buildah CLI.

In short, they'd like the administrator to be able to disallow the use of the `buildah from` command from the command line..

**Comment by sachinkaushik** [comment]
Thank you @TomSweeneyRedHat @rhatdan !

FYI..

I followed below tutorial for creating rootless openshitft container.  We want build user to block/disable buildah from command.  If build user try to do so, there should be message on cli that "This command is not allowed " or we can customize it accordingly. 

https://github.com/containers/buildah/blob/master/docs/tutorials/05-openshift-rootless-bud.md
